samples: nrf9160: modem_shell: External cloud support for location (#9804)
nrfconnect/sdk-nrf: samples/nrf9160/modem_shell/README.rst

See :ref:`lib_location` library for information on the configuration of different location methods and services.
Some default configurations are available to facilitate trials.

> The MoSh sample is using cloud service for positioning through Location library by default.

Reviewer: ```suggestion
This sample is using cloud service for positioning through the :ref:`lib_location` library by default.
```

> However, application can also handle the cloud communication for the location services by itself.

Reviewer: ```suggestion
However, an application can also handle the cloud communication for the location services by itself.
```

> However, most common use case is that application handles the cloud communication for the location services by itself.

Author: I've removed "most common use case".
That has originated from discussions with Norway teams when they've told it. So it's not just about ATv2. The statement meant end products but it's definitely true that for getting started with development and trials, user should let Location library handle the cloud communication.
Anyway, there are no remarks about what is common and what is not anymore.

> This can be enabled with :kconfig:option:`CONFIG_LOCATION_SERVICE_EXTERNAL` Kconfig option and then
> a separate configuration (``overlay-cloud_mqtt.conf``) must be used to enable nRF Cloud service over MQTT for retrieving location data.
> ``cloud`` command must be used to establish the MQTT connection before ``location`` commands.

Reviewer: ```suggestion
To enable cloud communication, use the :kconfig:option:`CONFIG_LOCATION_SERVICE_EXTERNAL` Kconfig option and 
a separate configuration (:file:`overlay-cloud_mqtt.conf`)  to enable nRF Cloud service over MQTT for retrieving location data.
``cloud`` command must be used to establish the MQTT connection before ``location`` commands.
```

> ``cloud`` command must be used to establish the MQTT connection before ``location`` commands.

Reviewer: ```suggestion
Use the ``cloud`` command to establish the MQTT connection before ``location`` commands.
```
Sorry, I missed this in my first review round.

   west build -p -b nrf9160dk_nrf9160_ns -d build -- -DOVERLAY_CONFIG=overlay-cloud_mqtt.conf

Location service handled in application
=======================================

> The MoSh sample is using cloud service for positioning through Location library by default.

Reviewer: ```suggestion
This sample is using cloud service for positioning through the :ref:`lib_location` library by default.
```

> To build the MoSh sample with location cloud services handled in the MoSh,
> use the ``-DOVERLAY_CONFIG="overlay-cloud_mqtt.conf"`` and ``-DCONFIG_LOCATION_SERVICE_EXTERNAL=y`` options.

Reviewer: ```suggestion
To build the sample with location cloud services handled in the MoSh,
use the ``-DOVERLAY_CONFIG="overlay-cloud_mqtt.conf"`` and ``-DCONFIG_LOCATION_SERVICE_EXTERNAL=y`` options.
```
